core/capabilities/ccip/ccipsolana/pluginconfig.go

```diff
@@ -0,0 +1,35 @@
+package ccipsolana
+
+import (
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ocrimpls"
+
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// PluginConfig is a struct that contains the configuration for a plugin.
+type PluginConfig struct {
+	extraDataCodec ccipcommon.ExtraDataCodec
+}
```

Comment:
I'm having some trouble following this. Don't we already have a `plugin` type in `oraclecreator`?

Reply:
in the oraclecreator/plugin.go, Blaž introduced a map that includes [this](https://github.com/smartcontractkit/chainlink/blob/f2c7e3a08db294a73f972a4e45ad00653f093207/core/capabilities/ccip/oraclecreator/plugin.go#L96) type, and it's initialized inside the oraclecreator and [stored](https://github.com/smartcontractkit/chainlink/blob/f2c7e3a08db294a73f972a4e45ad00653f093207/core/capabilities/ccip/oraclecreator/plugin.go#L65) in a map